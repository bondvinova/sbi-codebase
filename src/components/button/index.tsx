import * as React from 'react';
import { Slot } from '@radix-ui/react-slot';
import { cva, type VariantProps } from 'class-variance-authority';

import { cn } from '@/lib/clsx-twmerge';

export const buttonVariants = cva(
  'inline-flex items-center justify-center rounded-sm focus-visible:outline-none focus-visible:ring-2 disabled:pointer-events-none active:scale-[0.98] transition-all duration-300 outline-none tracking-normal text-white text-base font-semibold',
  {
    variants: {
      variant: {
        default:
          'bg-sbi-blue-100 disabled:bg-blue-30 focus:ring focus:ring-sbi-blue-100/50 focus:ring-offset-1 hover:bg-sbi-blue-100/90 focus-visible:ring focus-visible:ring-offset-1 focus-visible:ring-sbi0-blue-100/50',
        outline: 'border border-sbi-blue-100 bg-white text-sbi-blue-100',
      },
      size: {
        default: 'h-10 px-4 py-2',
        sm: 'h-8 px-3 py-2',
      },
    },
    defaultVariants: {
      variant: 'default',
      size: 'default',
    },
  }
);

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
    VariantProps<typeof buttonVariants> {
  asChild?: boolean;
}

const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  ({ className, variant, size, asChild = false, ...props }, ref) => {
    const Comp = asChild ? Slot : 'button';
    return <Comp className={cn(buttonVariants({ variant, size, className }))} ref={ref} {...props} />;
  }
);
Button.displayName = 'Button';

export default Button;
