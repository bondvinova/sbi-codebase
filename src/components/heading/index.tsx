import React, { HTMLAttributes } from 'react';
import { cva, VariantProps } from 'class-variance-authority';

import { cn } from '@/lib/clsx-twmerge';

const headingVariants = cva('m-0 font-semibold', {
  variants: {
    variant: {
      'title-form': 'text-blue-80',
      default: 'text-grey-slate-110',
    },
    size: {
      default: 'text-2xl leading-8',
      sm: 'text-base leading-6',
    },
  },
  defaultVariants: {
    variant: 'default',
    size: 'default',
  },
});

interface IHeadingProps extends HTMLAttributes<HTMLHeadingElement>, VariantProps<typeof headingVariants> {
  type?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
}

const Heading = React.forwardRef<HTMLHeadingElement, IHeadingProps>(
  ({ type = 'h3', className, variant, size, ...props }, ref) => {
    const Type = type;
    return (
      <Type
        {...props}
        className={cn(
          headingVariants({
            variant,
            size,
            className,
          })
        )}
        ref={ref}
      />
    );
  }
);

Heading.displayName = 'Heading';

export default Heading;
