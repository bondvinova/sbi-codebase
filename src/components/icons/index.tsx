import { useState } from 'react';

import Button from '@/components/button';
import Heading from '@/components/heading';
import AlertIcon from '@/components/icons/AlertIcon';
import SuccessIcon from '@/components/icons/SuccessIcon';
import Input from '@/components/input';
import { cn } from '@/lib/clsx-twmerge';
import type { KeyOfIcon, ValueOfIcon } from '@/types/icons';

export const IconsObject = {
  AlertIcon,
  SuccessIcon,
};

const Icons = () => {
  const [search, setSearch] = useState('');

  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
    setSearch(e.target.value);
  };

  const array = (Object.entries(IconsObject) as [KeyOfIcon, ValueOfIcon][]).filter(([iconName]) => {
    return iconName.toLowerCase().includes(search.toLowerCase());
  });

  return (
    <div className="flex flex-col gap-4">
      <Input placeholder="Filter icon by name or tag" value={search} onChange={handleSearch} />
      <p>Click on an icon to copy the name</p>
      {array.length > 0 ? (
        <div className="flex flex-wrap justify-center gap-4">
          {array.map(([iconName, Icon]) => (
            <IconItem Icon={Icon} key={iconName} iconName={iconName} />
          ))}
        </div>
      ) : (
        <p className="flex-1 text-center text-gray-500">No Icon found!</p>
      )}
    </div>
  );
};

interface IconItemProps extends React.HTMLAttributes<HTMLDivElement> {
  Icon: ValueOfIcon;
  iconName: KeyOfIcon;
}

const IconItem = ({ Icon, iconName }: IconItemProps) => {
  async function copyTextToClipboard(name: KeyOfIcon) {
    if ('clipboard' in navigator) {
      return await navigator.clipboard.writeText(`import ${name} from '@/components/icons/${name}'`);
    } else {
      return document.execCommand('copy', true, name);
    }
  }

  const handleCopyClick = (name: KeyOfIcon) => {
    copyTextToClipboard(name);
  };

  return (
    <Button className={cn('h-24 w-[160px] flex-col')} onClick={() => handleCopyClick(iconName)} variant="outline">
      <div className="flex p-2">
        <Icon />
      </div>
      <Heading className="text-sbi-blue-100" size="sm">
        {iconName}
      </Heading>
    </Button>
  );
};

export default Icons;
