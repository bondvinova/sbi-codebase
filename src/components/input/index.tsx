import * as React from 'react';

import { cn } from '@/lib/clsx-twmerge';

export interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  errorMessage?: string;
}

const Input = React.forwardRef<HTMLInputElement, InputProps>(
  ({ className, type, errorMessage, disabled, ...props }, ref) => {
    return (
      <input
        type={type}
        className={cn(
          'placeholder:text-grey-slate-70 flex h-12 w-full rounded-sm border border-grey-slate-30 bg-transparent bg-white px-4 py-3 text-base leading-6 text-grey-slate-110 outline-none transition-all duration-300 file:border-0 file:bg-transparent file:text-sm file:font-medium hover:border-sbi-blue-100 focus:border-sbi-blue-100 focus:shadow-form focus:outline-none focus-visible:shadow-form focus-visible:outline-none',
          {
            'focus:shadow-form-error focus-visible:shadow-form-error border-danger-70 hover:border-danger-70 focus:border-danger-70':
              errorMessage,
            'cursor-not-allowed hover:border-blue-30': disabled,
          },
          className
        )}
        ref={ref}
        disabled={disabled}
        {...props}
      />
    );
  }
);
Input.displayName = 'Input';

export default Input;
