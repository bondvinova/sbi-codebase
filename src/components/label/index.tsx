import * as React from 'react';
import * as LabelPrimitive from '@radix-ui/react-label';
import { cva, type VariantProps } from 'class-variance-authority';

import { cn } from '@/lib/clsx-twmerge';

const labelVariants = cva('peer-disabled:cursor-not-allowed peer-disabled:opacity-70 font-normal', {
  variants: {
    variant: {
      default: 'text-global-slate-80',
      select: 'text-grey-slate-110',
    },
    size: {
      default: 'text-sm leading-5',
      base: 'text-base leading-6',
    },
  },
  defaultVariants: {
    variant: 'default',
    size: 'default',
  },
});

export interface LabelProps extends React.LabelHTMLAttributes<HTMLLabelElement>, VariantProps<typeof labelVariants> {}

const Label = React.forwardRef<
  React.ElementRef<typeof LabelPrimitive.Root>,
  React.ComponentPropsWithoutRef<typeof LabelPrimitive.Root> & LabelProps
>(({ className, variant, size, ...props }, ref) => (
  <LabelPrimitive.Root ref={ref} className={cn(labelVariants({ variant, size }), className)} {...props} />
));
Label.displayName = LabelPrimitive.Root.displayName;

export default Label;
