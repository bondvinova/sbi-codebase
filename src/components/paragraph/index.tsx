import React, { HTMLAttributes } from 'react';
import { cva, VariantProps } from 'class-variance-authority';

import { cn } from '@/lib/clsx-twmerge';

const paragraphVariants = cva('max-w-prose text-slate-700 dark:text-slate-300 mb-2 text-center', {
  variants: {
    size: {
      default: 'text-base sm:text-lg',
      sm: 'text-sm sm:text-base',
    },
  },
  defaultVariants: {
    size: 'default',
  },
});

interface IParagraphProps extends HTMLAttributes<HTMLParagraphElement>, VariantProps<typeof paragraphVariants> {}

const Paragraph = React.forwardRef<HTMLParagraphElement, IParagraphProps>((props, ref) => {
  const { className, size, children, ...rest } = props;

  return (
    <p ref={ref} {...rest} className={cn(paragraphVariants({ size, className }))}>
      {children}
    </p>
  );
});

Paragraph.displayName = 'Paragraph';

export default Paragraph;
