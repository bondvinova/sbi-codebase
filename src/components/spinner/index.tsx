import React from 'react';

import { cn } from '@/lib/clsx-twmerge';

interface SpinnerProps extends React.HTMLAttributes<HTMLDivElement> {}

const Spinner = React.forwardRef<HTMLDivElement, SpinnerProps>((props, ref) => {
  const { className, ...rest } = props;

  return (
    <div className={cn('flex h-screen flex-1 flex-col items-center justify-center', className)} ref={ref} {...rest}>
      <div className="h-12 w-12 rounded-full border-[3px] border-transparent">
        <div className="animate-rotate absolute h-12 w-12 rounded-full border-[3px] border-transparent border-l-orange-400"></div>
        <div className="animate-rotateOpacity1 absolute h-12 w-12 rounded-full border-[3px] border-transparent border-l-orange-400"></div>
        <div className="animate-rotateOpacity2 h-12 w-12 rounded-full border-[3px] border-transparent border-l-orange-400"></div>
      </div>
    </div>
  );
});

Spinner.displayName = 'Spinner';

export default Spinner;
