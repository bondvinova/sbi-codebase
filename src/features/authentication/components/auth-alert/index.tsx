import React from 'react';
import { cva, type VariantProps } from 'class-variance-authority';

import { cn } from '@/lib/clsx-twmerge';

export const authAlertVariants = cva(
  'flex w-full items-center justify-start space-x-2 rounded-sm p-4 text-sm leading-5 text-global-slate-100 border',
  {
    variants: {
      variant: {
        danger: 'border-danger-60 bg-danger-10',
        warning: 'border-warning-60 bg-warning-10',
      },
    },
    defaultVariants: {
      variant: 'danger',
    },
  }
);

interface AuthAlertProps extends React.HTMLAttributes<HTMLDivElement>, VariantProps<typeof authAlertVariants> {
  renderIcon?: () => JSX.Element;
}

const AuthAlert = React.forwardRef<HTMLDivElement, AuthAlertProps>((props, ref) => {
  const { className, renderIcon, variant, children, ...rest } = props;
  return (
    <p ref={ref} className={cn(authAlertVariants({ variant }), className)} {...rest}>
      {renderIcon?.()}
      <span>{children}</span>
    </p>
  );
});

AuthAlert.displayName = 'AuthAlert';

export default AuthAlert;
