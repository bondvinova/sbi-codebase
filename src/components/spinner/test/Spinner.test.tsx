import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Spinner } from '@/components/spinner';

describe('Spinner component', () => {
  it('Should render and match the snapshot', () => {
    const { baseElement } = render(<Spinner />);

    expect(baseElement).toMatchSnapshot();
  });
});
