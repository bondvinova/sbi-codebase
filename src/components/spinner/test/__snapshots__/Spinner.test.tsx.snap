// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Spinner component > Should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="flex h-screen flex-1 flex-col items-center justify-center"
    >
      <div
        class="h-12 w-12 rounded-full border-[3px] border-transparent"
      >
        <div
          class="animate-rotate absolute h-12 w-12 rounded-full border-[3px] border-transparent border-l-orange-400"
        />
        <div
          class="animate-rotateOpacity1 absolute h-12 w-12 rounded-full border-[3px] border-transparent border-l-orange-400"
        />
        <div
          class="animate-rotateOpacity2 h-12 w-12 rounded-full border-[3px] border-transparent border-l-orange-400"
        />
      </div>
    </div>
  </div>
</body>
`;
